xcodeproj/internal/input_files.bzl

```diff
@@ -390,7 +390,7 @@ def _collect_input_files(
         }
 
         transitive_extra_files.extend([
-            d
+            depset(transitive = [depset([(label, d)])])
```

Comment:
```suggestion
            depset([(label, d)])
```

Reply:
I (or you? 🙏) can add an example into `example/rules_ios` that exercises this (in another if needed).

Reply:
If this can wait I can def add that Mon, I used `transitive` trying to take other code paths as an example but without fully understanding what I was doing haha 

But adding the test case and incorporating your change will help me educated myself here, thx!

Reply:
Yep, it can wait 👍.

Reply:
Until then If this blocks anyone working on `main` feel free to land and I’ll add the test case in a separate PR, otherwise I’d prefer to add everything here

Reply:
@brentleyjones let me know what you think, I can add more resources here if we want to exercise other variations but from a project generation perspective as long as we're correctly taking it into consideration I think it's enough but I could be wrong

Reply:
FYI I'll squash and sign off a single commit once we're good with the changes